Kyle/v0.3

kyle-stytch opened the pull request:
A draft of the complete overhaul of the SDK I did, includes SMS Passcode UI work. Not in a finished state, though it's pretty close. Various small-ish things to clean up, and some improvements to handling error messages, *and* I need to add tests, but otherwise I'm pretty happy with the state it's in. But feedback certainly welcome about anything and everything! Also any questions you have feel free to leave a comment or ping me about on slack.

ethan-stytch commented on sdk/src/main/java/com/stytch/sdk/StytchUI.kt:
```diff
@@ -1,107 +1,233 @@
-        color = StytchColor.fromColorId(R.color.editTextColor)
-        size = 16.dp
-    }
+        public abstract class Authenticator {
+            internal lateinit var callback: (Boolean) -> Unit
+
+            public fun onComplete(success: Boolean) {
+                callback(success)
+            }
```
We can discuss about best ways to go about this on both platforms.

kyle-stytch commented on sdk/src/main/java/com/stytch/sdk/Stytch.kt:
```diff
@@ -1,138 +1,42 @@
-    override fun toString(): String {
-        return "StytchResult(userId='$userId', requestId='$requestId')"
-    }
+internal object StytchLog {
+    fun e(message: String) = Log.e("StytchLog", "Stytch error: $message")
+    fun w(message: String) = Log.w("StytchLog", "Stytch warning: $message")
+    fun i(message: String) = Log.i("StytchLog", message)
+    fun d(message: String) = Log.d("StytchLog", message)
+    fun v(message: String) = Log.v("StytchLog", message)
```
e = error, w = warning, i = info, d = debug, v = verbose. Standard android logging levels.

kyle-stytch commented on sdk/src/main/java/com/stytch/sdk/StytchApi.kt:
```diff
@@ -0,0 +1,324 @@
+    )
+}
+
+public object StytchRequestTypes {
+    @JsonClass(generateAdapter = true)
+    public data class LoginOrCreateUserByEmailRequest(
+        val email: String,
+        val login_magic_link_url: String,
+        val signup_magic_link_url: String,
+        val login_expiration_minutes: Int?,
+        val signup_expiration_minutes: Int?,
+        val create_user_as_pending: Boolean?,
+        val attributes: StytchDataTypes.Attributes?,
+    )
+
+    @JsonClass(generateAdapter = true)
+    public data class LoginOrCreateUserBySMSRequest(
+        val phone_number: String,
+        val expiration_minutes: Int?,
+        val create_user_as_pending: Boolean?,
+        val attributes: StytchDataTypes.Attributes?,
+    )
+
+    @JsonClass(generateAdapter = true)
+    public data class LoginOrCreateUserByWhatsAppRequest(
+        val phone_number: String,
+        val expiration_minutes: Int?,
+        val create_user_as_pending: Boolean?,
+        val attributes: StytchDataTypes.Attributes?,
+    )
+}
+
```
I'm not sure what you mean, but I'd be happy to discuss.

ethan-stytch commented on sdk/src/main/java/com/stytch/sdk/Stytch.kt:
```diff
@@ -1,138 +1,42 @@
-        checkIfConfigured()
-        flowManager.resendEmailVerification()
+    @JvmStatic
+    public fun configure(
+        publicToken: String,
+        environment: StytchEnvironment,
+    ) {
```
Are we configuring without the projectID, secret, scheme, and host? Or through the deep linking path?

kyle-stytch commented:
@ethan-stytch With the recent changes I've verified that this is in a working state. I also fixed the scrolling issues I talked about with you (and moved the "Continue" button up so that it is no longer anchored to the bottom of the page).

I'm ready to merge this now as soon as you give your approval. I expect to put up another PR this weekend just to add a little more polish for Monday but, as you said, the main thing is getting this working and I have done that now.
